src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -147,6 +147,11 @@ class AsyncExceptionHandshake : public AsyncHandshakeClosure {
   }
 
   ~AsyncExceptionHandshake() {
+    Thread* current = Thread::current();
+    if (current->is_Java_thread()) {
```

Comment:
Why verify if this is a JavaThread? The target is always a JavaThread, and the sender of the exception is also a JavaThread(JVM_StopThread and JvmtiEnv::StopThread). Even the handshake operation will assert if the handshaker is not a JT.

Reply:
Will investigate.

Reply:
JavaThread::send_async_exception() creates a new AsyncExceptionHandshake.
JavaThread::send_async_exception() is called by JVM_StopThread() and
JvmtiEnv::StopThread(). The InstallAsyncExceptionHandshake which wraps
the AsyncExceptionHandshake is installed via Handshake::execute() which
calls "  JavaThread* self = JavaThread::current()" first thing so send_async_exception()
will fire an assert() if not called by a JavaThread.

Will fix.